platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpHandler.java

```diff
@@ -0,0 +1,285 @@
+                    createAuthnRequest(false), URLEncoder.encode(relayState, "UTF-8"));
+            idpRequest = idpMetadata.getSingleSignOnLocation() + "?" + queryParams;
+            UriBuilder idpUri = new UriBuilderImpl(new URI(idpRequest));
+
+            simpleSign.signUriString(queryParams, idpUri);
+
+            redirectUrl = idpUri.build().toString();
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.warn("Unable to encode relay state: " + relayState, e);
+            throw new ServletException("Unable to create return location");
+        } catch (SimpleSign.SignatureException e) {
+            String msg = "Unable to sign request";
+            LOGGER.warn(msg, e);
+            throw new ServletException(msg);
+        } catch (URISyntaxException e) {
+            LOGGER.warn("Unable to parse IDP request location: " + idpRequest, e);
+            throw new ServletException("Unable to determine IDP location.");
+        }
+
+        try {
+            response.sendRedirect(redirectUrl);
+            response.flushBuffer();
+        } catch (IOException e) {
+            LOGGER.warn("Unable to redirect AuthnRequest to " + redirectUrl, e);
+            throw new ServletException("Unable to redirect to IdP");
+        }
+    }
+
+    private void doHttpPostBinding(HttpServletRequest request, HttpServletResponse response) throws ServletException {
+        try {
+            response.getWriter().printf(postBindingTemplate, idpMetadata.getSingleSignOnLocation(), createAuthnRequest(true),
+                    createRelayState(request));
+            response.setStatus(200);
+            response.flushBuffer();
+        } catch (IOException e) {
+            LOGGER.warn("Unable to post AuthnRequest to IdP", e);
+            throw new ServletException("Unable to post to IdP");
+        }
+    }
+
+    private String createAuthnRequest(boolean isPost) throws ServletException {
+
+        String spIssuerId = String.format("https://%s:%s%s/saml", baseUrl.getHost(),
+                baseUrl.getHttpsPort(), baseUrl.getRootContext());
+        String spAssertionConsumerServiceUrl = spIssuerId + "/sso";
+
+        AuthnRequest authnRequest = authnRequestBuilder.buildObject();
+
+        Issuer issuer = issuerBuilder.buildObject();
+        issuer.setValue(spIssuerId);
+        authnRequest.setIssuer(issuer);
+
+        authnRequest.setAssertionConsumerServiceURL(spAssertionConsumerServiceUrl);
+
+        authnRequest.setID("_" + UUID.randomUUID().toString());
+        authnRequest.setVersion(SAMLVersion.VERSION_20);
+        authnRequest.setIssueInstant(new DateTime());
+
+        authnRequest.setDestination(idpMetadata.getSingleSignOnLocation());
+
+        authnRequest.setProtocolBinding(idpMetadata.getSingleSignOnBinding());
+        authnRequest.setNameIDPolicy(SamlpRequestComponentBuilder
+                .createNameIDPolicy(true, "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent",
```

Comment:
I'd prefer to use SAML2Constants.NAMEID_FORMAT_PERSISTENT than the string itself.